tests/publiccloud/check_registercloudguest.pm

```diff
@@ -106,14 +106,14 @@ sub run {
     if (is_byos()) {
         $instance->ssh_assert_script_run(cmd => 'sudo SUSEConnect --version');
         $instance->ssh_assert_script_run(cmd => "sudo SUSEConnect $regcode_param");
-        $instance->ssh_assert_script_run(cmd => "sudo ${path}registercloudguest --clean");
+        cleanup_instance($instance);
```

Comment:
You added here a function in place of a single command line: IMHO it is not much useful for the code visibility and simplicity: I suggest minimal changes, where needed.